### DISCUSSION TITLE
Cannot set net8.0 as first in TargetFrameworks list (SDK5.2, single app)

### DISCUSSION THREAD

**Discussion by AndyCW** [unoplatform/uno#16749] [General]
I have just migrated an app to Uno SDK5.2 and the Single App architecture. It builds and runs on WinSDK, Windows and Linux desktop , so far so good.

But I have the following Build warning:
```1>C:\Users\andy.wigley\.nuget\packages\uno.sdk\5.2.121\targets\Uno.Sdk.After.targets(77,3): warning UNOB0011: The desktop TargetFramework must not be placed first in the TargetFrameworks property in order for other platforms debugging to work. (See https://aka.platform.uno/UNOB0011)```

And this is because in my project file, I have:
```
  <TargetFrameworks>
        net8.0-desktop;
        net8.0;
        net8.0-windows10.0.19041;
   </TargetFrameworks>
```

So if I put **net8.0-windows10.0.19041** first in the list, I get the same build warning, but this time for the Windows target. So the obvious solution is to put .net8.0 first:
```
  <TargetFrameworks>
        net8.0;
        net8.0-desktop;
        net8.0-windows10.0.19041;
   </TargetFrameworks>
```

But this messes up the build and corrupts the Frameworks list in the Visual Studio Debug Targets dropdown. It seems to parse the TargetFrameworks element incorrectly and decides that I have asked for a framework named 'net8.0-windows10.0.19041-desktop':
```
1>------ Build started: Project: McStubUno, Configuration: Debug Any CPU ------
NuGet package restore failed. Please see Error List window for detailed warnings and errors.
Error occurred while restoring NuGet packages: Invalid restore input. Invalid target framework 'unsupported'. Input files: C:\Data\Bitbucket\arc-mc-stub-uno\McStubUno\McStubUno.csproj.
1>C:\Program Files\dotnet\sdk\8.0.205\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.TargetFrameworkInference.targets(96,5): error NETSDK1013: The TargetFramework value 'net8.0-windows10.0.19041-desktop' was not recognized. It may be misspelled. If not, then the TargetFrameworkIdentifier and/or TargetFrameworkVersion properties must be specified explicitly.
1>Done building project "McStubUno.csproj" -- FAILED.
```

So, any ideas on how to resolve this would be most welcome.

**Comment by jeromelaban** [comment]
Thanks for the report. Is `McStubUno` a class library or an app head? If it's a library, we have a fix that will be released to today to address that problem.

**Reply by jeromelaban** [reply]
Thanks for the response. I'm not able to reproduce this issue. Could you provide a [binlog of the error](https://platform.uno/docs/articles/uno-builds-troubleshooting.html)?

**Reply by jeromelaban** [reply]
@AndyCW, I'm not seeing binlog, could you add it directly from the github interface?

**Comment by AndyCW** [comment]
Thanks for quick response.
It is an app head.
________________________________

**Comment by AndyCW** [comment]
binlog....
________________________________

**Reply by AndyCW** [reply]
[McStubUno_Debug_AnyCPU_Build_2024-05-16T15_10_16.3954228+02_00.zip](https://github.com/unoplatform/uno/files/15336404/McStubUno_Debug_AnyCPU_Build_2024-05-16T15_10_16.3954228%2B02_00.zip)

**Reply by jeromelaban** [reply]
Thank you. 

The source of the error does not appear in that binlog alone. Could you also send all the other binlogs that are in the logs window? Could you also try building the same project from the command line and send the binlog as well?

**Reply by AndyCW** [reply]
If I use `dotnet build /bl` there is no error. This error only happens building from Visual Studio.

Here are more logs:
[McStubUno_Debug_AnyCPU_Build_2024-05-16T16_55_02.3666233+02_00.zip](https://github.com/unoplatform/uno/files/15337104/McStubUno_Debug_AnyCPU_Build_2024-05-16T16_55_02.3666233%2B02_00.zip)

Here is the command line binlog: 
[msbuild-cli.zip](https://github.com/unoplatform/uno/files/15337121/msbuild-cli.zip)

**Reply by jeromelaban** [reply]
Thanks. 

The command line shows it's a VS specific issue. For the VS logs, I'm interested in the ones that are marked "design time", those are the ones that may mangle the target frameworks. To best have them, open VS without opening the solution, enable the VS logging, then open the solution. Let VS settle down, the build once. Take all the binlogs and post them here, thanks!

**Reply by AndyCW** [reply]
binlogs: 
[McStubUno_Debug_AnyCPU_net8.0-windows10.0.19041-desktop_DesignTimeBuild_2024-05-17T12_58_11.4759514+02_00.zip](https://github.com/unoplatform/uno/files/15348080/McStubUno_Debug_AnyCPU_net8.0-windows10.0.19041-desktop_DesignTimeBuild_2024-05-17T12_58_11.4759514%2B02_00.zip)

**Reply by jeromelaban** [reply]
Thanks, this helped.

We'll be providing a fix soon.

Note that this is caused by the presence of `net8.0` which is enabled when adding support for unit tests.

**Comment by AndyCW** [comment]
Added.
________________________________
